deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/worker/HealthChecker.java

```diff
@@ -440,20 +440,18 @@ private void processCompletingState(HealthCheckBean healthCheckBean, GroupBean g
             LOG.info("The host {} has been terminated", hostId);
             healthCheckBean.setHost_terminated(true);
         } else {
-            HostBean hostBean = new HostBean();
             if (healthCheckBean.getStatus() == HealthCheckStatus.FAILED || healthCheckBean.getStatus() == HealthCheckStatus.TIMEOUT) {
                 LOG.info(String.format("Health check %s. Retain the host %s for debugging", healthCheckBean.getStatus().toString(), hostId));
-                hostBean.setState(HostState.PENDING_TERMINATE);
-                hostBean.setLast_update(System.currentTimeMillis());
                 healthCheckBean.setHost_terminated(false);
             } else {
                 LOG.info(String.format("Health check %s. Termimate the host %s", healthCheckBean.getStatus().toString(), hostId));
                 hostInfoDAO.terminateHost(hostId);
+                HostBean hostBean = new HostBean();
                 hostBean.setState(HostState.TERMINATING);
                 hostBean.setLast_update(System.currentTimeMillis());
+                hostDAO.updateHostById(hostId, hostBean);
```

Comment:
What I am thinking is, we can also move this logic (line 448 to 453) to the HealthCheckHostTerminator.java 
The extra work we need to add in the HealthCheckHostTerminator.java, is for each HealthCheckBean, we check the status:

if (healthCheckBean.getStatus() == HealthCheckStatus.SUCCEEDING || System.currentTimeMillis() - bean.getHost_launch_time() > timeToRetain) {
 // terminate instance.
}